### PR TITLE
Populate per ruleset statistics in `APIAccess` for use in difficulty recommender

### PR DESCRIPTION
**Pull request by LittleEndu**
Alternative to how it's done in #18498 

Based on the suggestion in #12898 by frenzibyte, this PR makes a new bindable in `IAPIProvider` that has local user's statistics per ruleset. That value is then used in difficulty recommender to make recommendations based on current ruleset pp.

### CODE REVIEW

File: osu.Game/Online/API/APIAccess.cs [hunk 1]
```diff
@@ -203,6 +207,13 @@ private void run()
                             continue;
                         }
 
+                        // populating ruleset statistics
+                        var usersRequest = new GetUsersRequest(new[] { localUser.Value.Id });
+
+                        usersRequest.Success += res => statistics.Value = res.Users.First().RulesetsStatistics;
```
Comment (smoogipoo) [inline osu.Game/Online/API/APIAccess.cs:213]: We already have a `GetUserRequest` above, does that not provide the ruleset statistics or something?

Comment (LittleEndu) [reply osu.Game/Online/API/APIAccess.cs:213]: It's only for one ruleset not all four. See the deleted part in difficulty recommender where get user was called 4 times

Comment (smoogipoo) [reply osu.Game/Online/API/APIAccess.cs:213]: In-fact it doesn't seem to contain any if you don't provide a ruleset? Neither the default ruleset or anything, seems kinda weird.

Comment (peppy) [reply osu.Game/Online/API/APIAccess.cs:213]: Checking with @nanaya , it's probably simple enough to change `/me` to return all rulesets, and use that instead? Would save extra API requests and complication.

**Comment by peppy** [comment]
APIv2 will now return all ruleset statistics in `/me` lookups. See https://github.com/ppy/osu-web/pull/8989/files for reference.

**Comment by LittleEndu** [comment]
* Closing in favour of https://github.com/ppy/osu/pull/18623

**Pull request closed without merging**